test-headless-cache.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
```

Comment:
This file is useful for manually validating the plaintext cache fallback, but I'd like to suggest it belongs in a `tests/` directory, not the repo root.